### DISCUSSION TITLE
J2534 (passthrucan) connections not possible in V203

### DISCUSSION THREAD

**Discussion by SHAUNINGTON** [collin80/SavvyCAN#327] [Q&A]
Hi All,

Have recently started using SavvyCAN for some RE projects - I am really enjoying it so far.
I have been using V199 but run into occasional crashes and things that don't quite work quite right (playback retaining original timing, for example)

Anyhow - I have downloaded V203, but when I try to connect a J2534 device, the 'Port' menu does not populate.
Please see attached screenshots.

Anything I am doing wrong?

![v199](https://user-images.githubusercontent.com/82267180/114267925-e32f1f00-9a41-11eb-97f7-bd6bcd851f99.png)

![v203](https://user-images.githubusercontent.com/82267180/114267928-e4604c00-9a41-11eb-9357-4f677557225a.png)

**Comment by collin80** [comment]
The only thing I can think of is that V199 was compiled by me manually. After that I switched to CI/CD builds. So, perhaps those builds don't have all the proper plugins? I'm not sure, I'll have to try to look into that. In the meantime, compiling it yourself would probably give you a version with all the proper plugins.

**Reply by SHAUNINGTON** [reply]
Thanks @collin80 for the reply.
Would appreciate if you could look into it. I have a Korlan USB2CAN (the new product) and note it is not supported by your software (hopefully one day?) and a Drewtech J2534 was what I used initially.

I will have a go at doing a self compile (don't have the highest of hopes there) when I get some time.

**Reply by bmartins-eng** [reply]
I've run into something similar with passthru devices. Mine are all listed on the menu but disconnect when switching the baud rate to anything other than 250kBaud. Most of the vehicles I see are at 500kBaud which means I can't really use SavvyCAN with my current hardware. I'm new to this though so may be something I'm doing wrong!

**Reply by SHAUNINGTON** [reply]
Hi bmartins,

If you are using V199, I have noticed the J2534 implementation is a little buggy.
For example, this is my process in order to be able to capture or analyse traffic on a bus using J2534:

1. Open Savvy Can.
2. Connect CANH and CANL pins to bus in question
3. Open Connection Window
4. Click Add New Device Connection
5. Click QT SerialBus Devices Radio Button
6. Select J2534 interface being used
7. Click 'Create New Connection' - Default Baud Rate is 250, as you describe
8. Click to highlight the Connection (Often shows 'connected')
9. Change Baud Rate to 500
10. Click 'Save Bus Settings' (Status changes from connected to disconnected, and then nothing happens)
11. Click to highlight the connection again - press 'save bus settings' again 
12. Status usually says Not Connected, but if the connected bus is active, the traffic will start to flow for analysis.

I have to repeat this process each time using V199 x86.
I could not successfully get V20x to compile with J2534, but that could possibly be my fault.
It could also something to do with X64 (X20x) vs x86 (x199)? 
@collin80 if you have a chance to do a manual compile on a version, I'd love to give it a go.

@bmartins-eng let me know if that helps at all!

**Reply by bmartins-eng** [reply]
Thanks @SHAUNINGTON,

I thought it might have been me but it's reassuring to know it's not just my newbie status! I will give it a go using your above method. If still struggling then I'll have to spend some pennies on getting another device to make the process a little easier. 

Thanks again and I'll keep you posted.

Kind regards

Ben

**Reply by collin80** [reply]
I probably messed up the coding for passthrough devices. I've done very little testing with them so the code is not very well tested. This is something I'll have to fix when I get a chance.

**Reply by bmartins-eng** [reply]
Hi @collin80, really not a problem and I shouldn't be so tight and just go a buy a dedicated dongle which will work. Got my eye on a PeakCAN at the moment! It would be great though if you could get it to work with PassThru devices. I had it working fine my NexIq on J1939 running at 250kBaud but as soon as you change the baud rate to 500 it drops the connection and computer says no. I attempted what @SHAUNINGTON mentioned but sadly still no joy. If you do get a chance this would open up a lot more users to Savvy CAN. Thanks for every so far though @collin80, the software is a joy to use, keep up the good work.

**Reply by SHAUNINGTON** [reply]
> I probably messed up the coding for passthrough devices. I've done very little testing with them so the code is not very well tested. This is something I'll have to fix when I get a chance.

Hi Colin,
I’m sure you’ve had other priorities but any luck with this?

I’ve just clicked over to 207 latest x64 build and found still no J2534 devices to select from. As it stands, 199 x86 is the last version where this works - but it’s very crash happy and missing so much of your good work.

I do log files in v199 and then open in v204+ to analyse but it’s a bit iffy.

The passthrucan menu is there and implemented, but there are no devices to chose from (6 j2534 show up in v199)


Regards,
Shaun

**Reply by FLcruising** [reply]
Was a workaround ever resolved? My J2534 (Tactrix OP) doesn't work either.

**Comment by testuser7** [comment]
I think that the problem is in the QT framework.
Version V199.1 with QT 5.12.3 is working. Next version 1.0.20 with QT 5.15.0 is not working.
The same problem with passthrucan is observed in CANdevStudio application.

Manually compiled latest version with QT 5.15 is not working correctly.

**Reply by FLcruising** [reply]
Thank you! **V199.1** works!

**Comment by SHAUNINGTON** [comment]
As a work around I’ve been using V199 to capture and save but v204 or v207 to analyse. It works ok but if there is a way to get the pass through working on the later builds it would be a great benefit I feel On 21 Nov 2022, at 12:23 am, FLcruising ***@***.***> wrote:﻿
Thank you! V199.1 works!

**Reply by FLcruising** [reply]
Same here but not ideal

**Comment by testuser7** [comment]
Version V199.1 is 32-bit application. All next versions are 64-bit. Most j2543 drivers are 32-bit. You can try to use 64-bit driver for OpenPort 2.0: https://github.com/dschultzca/j2534/releases/tag/3.0.0 with this registry: https://pastebin.com/TjhWptWN
